FWI_forward_adjoint/FWIpart1.md

```diff
@@ -56,65 +56,68 @@ create highly optimized low-level C code. For this purpose Devito uses
 the symbolic algebra package SymPy [@Meurer17] to facilitate the
 automatic creation of derivative expressions, allowing the quick and
 efficient generation of high-order wave propagators with variable
-stencil orders.
+stencil orders, while automated performance optimization using
+state-of-the-art compiler techniques are used to ensure
+computational efficiency.
 
 At the core of Devito's symbolic API are two symbolic types that
 behave like SymPy function objects, while also managing
 user data:
 
-* `DenseData` objects represent a spatially varying function
+* `Function` objects represent a spatially varying function
   discretized on a regular cartesian grid. For example, a function
-  symbol `f = DenseData(name='f', shape=(nx, ny), space_order=2)`
+  symbol `f = Function(name='f', ..., space_order=2)`
   is denoted symbolically as `f(x, y)`. Auto-generated symbolic
   expressions for finite difference derivatives are provided by these
-  objects via shorthand expressions, enabling the syntax `f.dx` to
-  denote $\frac{\partial f}{\partial x}$ and `f.dx2` for
-  $\frac{\partial^2 f}{\partial x^2}$.
+  objects via shorthand expressions, where the `space_order` parameter
+  defines the order of the created finite difference stencil.
 
-* `TimeData` objects represent a time-dependent function
+* `TimeFunction` objects represent a time-dependent function
   that includes leading dimension $t$, for example `g(t, x, y)`. In
-  addition to spatial derivatives `TimeData` symbols also provide
-  time derivatives `g.dt` and `g.dt2`, as well as options to save
+  addition to spatial derivatives `TimeFunction` symbols also provide
+  time derivatives `g.dt` and `g.dt2`, as well as options to store
   the entire data along the time axis.
 
 To demonstrate Devito's symbolic capabilities, let us consider a
 time-dependent function $\vd{u}(t, x, y)$ representing the discrete forward
 wavefield. We can define this as a `TimeData` object in Devito:
 
 ```python
-    u = TimeData(name="u", shape=model.shape_domain, time_order=2,
-                 space_order=2, save=True, time_dim=nt)
+  u = TimeData(name='u', grid=model.grid, time_order=2,
+               space_order=2, save=True, timd_dim=nt)
 ```
 
-where the parameter `shape` defines the size of the allocated memory
-region, `time_order` and `space_order` define the default
-discretization order of the derivative expressions and the parameters
-`save=True` and `time_dim` force the entire wavefield to be stored in
-memory.
+where the the `grid` object provided by the `model` defines the size
+of the allocated memory region, `time_order` and `space_order` define
```

Comment:
notpicking,but maybe it would be better to introduce `grid` at the `model` part t oavoid the `...` above.